rust/rpmostree-client/src/lib.rs

```diff
@@ -36,16 +56,63 @@ pub struct Deployment {
     pub version: Option<String>,
 }
 
+impl Status {
+    /// Find the booted deployment, if any.
+    pub fn find_booted(&self) -> Option<&Deployment> {
+        self.deployments.iter().find(|d| d.booted)
+    }
+
+    /// Find the booted deployment.
+    pub fn require_booted(&self) -> Result<&Deployment> {
+        self.find_booted()
+            .ok_or_else(|| format!("No booted deployment").into())
+    }
+}
+
+impl Deployment {
+    /// Find the base OSTree commit
+    pub fn get_base_commit(&self) -> &str {
+        self.base_checksum
+            .as_deref()
```

Comment:
I only recently learned about `as_deref()`; takes the `Option<String>` to `Option<&str>` and makes this combinator chain quite elegant IMO.